kyo-core/shared/src/main/scala/kyo/Channel.scala

```diff
@@ -64,6 +67,32 @@ object Channel:
                 }
             }
 
+        /** Puts elements into the channel as a batch, asynchronously blocking if necessary. Breaks batch up if it exceeds channel capacity.
+          *
+          * @param values
+          *   Chunk of elements to put
+          */
+        def putBatch(values: Chunk[A])(using Frame): Unit < (Abort[Closed] & Async) =
+            if values.isEmpty then ()
+            else
+                IO.Unsafe {
+                    var i = 0
+                    boundary[Unit < (Abort[Closed] & Async)]:
+                        values.foreach: value =>
+                            self.offer(value) match
+                                case Result.Success(true) =>
+                                    i += 1
+                                case Result.Success(false) =>
+                                    boundary.break:
+                                        if i == values.length - 1 then
+                                            self.putFiber(values.last).safe.get
+                                        else
+                                            self.putBatchFiber(values.dropLeft(i)).safe.get
+                                        end if
+                                case Result.Fail(cl) =>
+                                    boundary.break(Abort.fail(cl))
```

Comment:
It's important to match on Panic as well. 

```suggestion
                                case e: Result.Error[?] =>
                                    boundary.break(Abort.get(e))
```

Reply:
Sorry, my comment was not clear. It's not necessary to match on both Panic and Fail, just match on the supertype `Error`. You can then use `Abort.get` to avoid re-allocating fail/panic classes.